walkthroughs/howto-tls-ecs-crossvpc/appmesh-baseline.yml

```diff
@@ -0,0 +1,951 @@
+            Retries: 3
+            StartPeriod: 10
+            Timeout: 2
+        - Name: envoy
+          Image: "840364872350.dkr.ecr.eu-west-2.amazonaws.com/aws-appmesh-envoy:v1.11.2.0-prod"
+          Essential: true
+          User: 1337
+          Environment:
+           - Name: "APPMESH_VIRTUAL_NODE_NAME"
+             Value: "mesh/appmesh-workshop/virtualNode/crystal-lb-vanilla"
+          HealthCheck:
+            Command:
+              - CMD-SHELL
+              - curl -s http://localhost:9901/server_info | grep state | grep -q LIVE
+            Retries: 3
+            Timeout: 2
+            Interval: 5
+            StartPeriod: 10
+      ProxyConfiguration:
+          ContainerName: envoy
+          Type: APPMESH
+          ProxyConfigurationProperties:
+          - Name: ProxyIngressPort
+            Value: "15000"
+          - Name: AppPorts
+            Value: "3000"
+          - Name: EgressIgnoredIPs
+            Value: "169.254.170.2,169.254.169.254"
+          - Name: IgnoredUID
+            Value: "1337"
+          - Name: ProxyEgressPort
+            Value: "15001"
+             
+  ExternalLoadBalancerSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupName: !Sub SecurityGroup-ExternalLoadBalancer-${AWS::StackName}
+      GroupDescription: Access to the external load balancer
+      VpcId: !Ref VPC
+      SecurityGroupIngress:
+          - IpProtocol: tcp
+            FromPort: 80
+            ToPort: 80
+            CidrIp: 0.0.0.0/0
+
+  ExternalLoadBalancer:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      Name: !Sub ExtLB-${AWS::StackName}
+      Scheme: internet-facing
+      Type: application
+      LoadBalancerAttributes:
+        - Key: idle_timeout.timeout_seconds
+          Value: '30'      
+      SecurityGroups:
+        - !Ref ExternalLoadBalancerSecurityGroup
+      Subnets:
+        - !Ref PublicSubnetOne
+        - !Ref PublicSubnetTwo
+        - !Ref PublicSubnetThree
+    DependsOn:
+      - PublicSubnetOneRouteTableAssociation
+      - PublicSubnetTwoRouteTableAssociation
+      - PublicSubnetThreeRouteTableAssociation
+
+  ExternalListener:
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    Properties:
+      DefaultActions:
+        - TargetGroupArn: !Ref RubyTargetGroup
+          Type: forward
+      LoadBalancerArn: !Ref ExternalLoadBalancer
+      Port: 80
+      Protocol: HTTP
+
+  RubyTargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Properties:
+      TargetType: instance
+      HealthCheckIntervalSeconds: 10
+      HealthCheckPath: /health
+      HealthCheckProtocol: HTTP
+      HealthCheckTimeoutSeconds: 5      
+      HealthyThresholdCount: 2
+      Matcher:
+        HttpCode: 200-299      
+      UnhealthyThresholdCount: 10
+      Port: 3000
+      Protocol: HTTP
+      VpcId: !Ref VPC
+      TargetGroupAttributes:
+        - Key: deregistration_delay.timeout_seconds
+          Value: '5'
```

Comment:
Generally speaking this value is too low and may cause issues with target health and availability, so we should steer away from it. The default (300s) is very safe, and I wouldn't go below 90s in most cases.